growthstreet_zoo/zoo.py

```diff
@@ -1,18 +1,15 @@
 """You are a zoo keeper. Write a set of objects that simulates a simple zoo."""
 import itertools
-from name_generator import generate_name
+from name_generator import GenerateNameMixin
 
 
-class Zoo(object):
+class Zoo(GenerateNameMixin):
     """Creates a Zoo object with a name and an empty list for cages."""
 
     def __init__(self, name=None):
         """Zoo is created with a name."""
         self.cages = []
-        if not name:
-            self.name = generate_name()
-        else:
-            self.name = name
+        self.name = self.generate_name(name)
```

Comment:
See if you can work out how to put this code in the mixin too.  (Hint: the __init__ method.)